### PR TITLE
feature/CP-7855-sdk-android-support-event-property-filters

### PR DESCRIPTION
**Pull request by unnaticleverpush**
Added feature in `AppBanner` to support event property filters in targeting section.

### CODE REVIEW

**Review by gbhrdt — APPROVED** [review]

File: cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java [hunk 1]
```diff
@@ -775,19 +775,54 @@ private boolean checkEventTriggerCondition(BannerTriggerCondition condition) {
...
+            if (bannerTrackEvents.isEmpty()) {
+              return false;
+            }
+
+            for (TableBannerTrackEvent bannerTrackEvent : bannerTrackEvents) {
+              int count = bannerTrackEvent.getCount();
+              String createdDate = bannerTrackEvent.getCreatedDateTime();
+              int pastDays = getDaysDifference(createdDate);
+
+              if (pastDays > Integer.parseInt(bannerTrackEvent.getProperty())) {
```
Comment (cpKevinrodriguesnobre) [inline cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java:809]: whats bannerTrackEvent.getProperty()?

**Review by cpKevinrodriguesnobre — CHANGES_REQUESTED** [review]

File: cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java (hunk 1 above)
Comment (unnaticleverpush) [reply cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java:809]: bannerTrackEvent is list that we fetch from local DB.

Screenshot 2024-06-11 at 4 00 14 PM

Screenshot 2024-06-11 at 3 51 00 PM

Comment (cpKevinrodriguesnobre) [reply cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java:809]: gotcha what happens for 0Days? and getDaysDifference(createdDate is same day?

Comment (unnaticleverpush) [reply cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java:809]: @cpKevinrodriguesnobre Can not add 0 in days.

Comment (cpKevinrodriguesnobre) [reply cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java:809]: well perfect i guess

**Review by cpKevinrodriguesnobre — APPROVED** [review]

**Pull request merged**